torchvision/prototype/features/_bounding_box.py

```diff
@@ -39,6 +39,7 @@ def __new__(
     def to_format(self, format: Union[str, BoundingBoxFormat]) -> BoundingBox:
         # TODO: this is useful for developing and debugging but we should remove or at least revisit this before we
         #  promote this out of the prototype state
+        # Do we still need this?
```

Comment:
Need, no. Want, yes. I actually want to keep them after this is promoted out of prototype. It is a convenient way to work with the features outside of a training pipeline. Compare for example

```py
from torchvision.prototype import features, transforms

transform = transforms.ConvertBoundingBoxFormat(format="xyxy")

bounding_box = features.BoundingBox(...)
xyxy_bounding_box = transform(bounding_box)
```

or

```py
from torchvision.prototype import features
from torchvision.prototype.transforms import functional as F

bounding_box = features.BoundingBox(...)
xyxy_bounding_box = features.BoundingBox.new_like(
    bounding_box, 
    convert_bounding_box_format(bounding_box, old_format=bounding_box.format, new_format="xyxy"),
    format="xyxy",
)
```

to

```py
from torchvision.prototype import features

bounding_box = features.BoundingBox(...)
xyxy_bounding_box = bounding_box.to_format("xyxy")
```

Still, I agree, the lazy import needs to go. My proposal is to move the actual conversion code from `transforms.functional` to a private function like `_convert_bounding_box_format` in this module. We could use it here in `to_format` and simply expose it in `transforms.functional`.

---

This comment applies to all other comments that ask the same.

Reply:
I agree that the verbosity of option 2 is an indication that we need a utility. It doesn't necessarily means that the method should be part of BoundingBox, so let's explore options. Note that we should avoid the use of strings in favour of their enums in our examples and internal code to get better static analysis. 

> This comment applies to all other comments that ask the same.

The 2 visualization methods `show()` and `draw_bounding_box()` on `Image` are not the same thing. We should remove them as we don't want to necessarily push PIL on the new API.

Reply:
>  Note that we should avoid the use of strings in favour of their enums in our examples and internal code to get better static analysis.

Internally I agree 100%, but do we actually want to discourage users from using them in application code? `"xyxy"` is more concise than `BoundingBoxFormat.XYXY` and potentially needs one less import.

Reply:
We have previously made the decision to move with enums (you can see this on multiple APIs including io, multi-weights etc). For consistency we should continue this here.

Where we agree is that people should be allowed to pass strings if they want to. I want to discourage them from doing so by offering all the examples an internal code using enums but if they choose to pass strings so be it. I just don't want to promote this usage. Note that this is true for virtually all places where we have enums.

Reply:
As discussed offline, quietly supporting strings is still a good idea, but everything else like documentation or examples we should use the enums.